### PR TITLE
feat: Add LicenseClient

### PR DESCRIPTION
**Pull request by allenchuang**

### CODE REVIEW

**Comment by allenchuang** [comment]
tests pass locally, probably need to add `LICENSING_MODULE_CONTRACT` to secrets

File: packages/core-sdk/src/resources/license.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+  /**
+   * List all licenses.
+   *
+   * @returns A Promise that resolves to the ListLicenseResponse.
+   */
+  public async list(): Promise<ListLicenseResponse> {
+    try {
+      const response: AxiosResponse = await this.httpClient.get(
+        `/license?franchiseId=7&ipAssetId=5`,
```
Comment (edisonz0718) [inline packages/core-sdk/src/resources/license.ts:59]: I think we should let users pass in the franchiseId and ipAssetId

**Review by edisonz0718 — COMMENTED** [review]
The test coverage is not 100%. Can you add unit tests to cover un-covered cases shown in the test coverage report?

File: packages/core-sdk/src/client.ts [hunk 2]
```diff
@@ -52,6 +54,23 @@ export class StoryClient {
     return this._franchise;
   }
 
+  /**
+   * Getter for the license client. The client is lazily created when
+   * this method is called.
+   *
+   * @returns the FranchiseClient instance
```
Comment (edisonz0718) [inline packages/core-sdk/src/client.ts:61]: Can you also add a unit test in client.test.ts to test this function

File: packages/core-sdk/src/resources/license.ts (hunk 1 above)
Comment (allenchuang) [reply packages/core-sdk/src/resources/license.ts]: Done! can you check again?

File: packages/core-sdk/src/client.ts (hunk 2 above)
Comment (allenchuang) [reply packages/core-sdk/src/client.ts:61]: added

**Review by edisonz0718 — APPROVED** [review]

**Pull request merged**